### PR TITLE
[api] Initial API deprecations for Ray 1.0

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Initial set of API changes for Ray 1.0, focusing on the core Ray APIs. Some of these parameter removals are breaking changes, but none of them should be particularly impactful.

Planning spreadsheet:
https://docs.google.com/spreadsheets/d/1F6VZ1R_fgDzpRZjXI5_6gaUwW2CQKi431dCDTR-Vm8s/edit#gid=0

### CODE REVIEW

**Comment by ericl** [comment]
cc @robertnishihara @pcmoritz @raulchen @richardliaw @stephanie-wang @edoakes @richardliaw

File: python/ray/tests/test_ray_init.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def test_redis_port(self, shutdown_only):
         def f():
             return 1
 
-        info = ray.init(redis_port=1234, redis_password="testpassword")
+        info = ray.init(redis_port=1234, _redis_password="testpassword")
```
Comment (stephanie-wang) [inline python/ray/tests/test_ray_init.py:66]: Ah was this a new decision? It actually seems a bit strange to me to have `redis_port` but then a hidden `_redis_password`.

File: doc/source/using-ray-with-gpus.rst [hunk 2]
```diff
@@ -1,4 +1,4 @@
-GPU Support
+tGPU Support
```
Comment (rkooo567) [inline doc/source/using-ray-with-gpus.rst:1]: Is t typo?

File: python/ray/tests/test_advanced_3.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def f():
 
 def test_specific_job_id():
     dummy_driver_id = ray.JobID.from_int(1)
-    ray.init(num_cpus=1, job_id=dummy_driver_id)
+    ray.init(num_cpus=1, _job_id=dummy_driver_id)
```
Comment (rkooo567) [inline python/ray/tests/test_advanced_3.py:265]: Job id seems to be in the deletion candidate? Are we keeping it for tests?

File: python/ray/tests/test_ray_init.py (hunk 1 above)
Comment (ericl) [reply python/ray/tests/test_ray_init.py:66]: We don't support redis_port any more; removing this test.

**Review by stephanie-wang — APPROVED** [review]

File: python/ray/serve/tests/test_router.py [hunk 4]
```diff
@@ -161,6 +161,7 @@ async def test_shard_key(serve_instance, task_runner_mock_actor):
             assert call.request_args[0] in runner_shard_keys[i]
 
 
+# TODO(fix)
```
Comment (ericl) [inline python/ray/serve/tests/test_router.py:164]: @edoakes @simon-mo this tests uses `ray.register_custom_serializer`, what's the best way to fix it?

Comment (simon-mo) [reply python/ray/serve/tests/test_router.py:164]: I'll remove it ASAP. on it

**Review by rkooo567 — APPROVED** [review]

File: python/ray/serve/tests/test_router.py (hunk 4 above)
Comment (simon-mo) [reply python/ray/serve/tests/test_router.py:164]: It's gone

**Review by edoakes — APPROVED** [review]
The catharsis is real! We should publish a summary of all of the breaking changes that are being made when we release 1.0.

**Comment by stephanie-wang** [comment]
I just realized that `num_redis_shards` isn't currently supported. I think it starts multiple redis shards but I don't think any of them except the primary are being used. Should we just remove that flag for now?

**Comment by ericl** [comment]
Looks like just 1 tune test failed; merging to unblock followup prs.

**Pull request merged**